solr/core/src/test/org/apache/solr/spelling/DirectSolrSpellCheckerTest.java

```diff
@@ -70,7 +71,7 @@ public void test() throws Exception {
               SpellingOptions spellOpts = new SpellingOptions(tokens, searcher.getIndexReader());
               SpellingResult result = checker.getSuggestions(spellOpts);
               assertNotNull("result shouldn't be null", result);
-              Map<String, Integer> suggestions = result.get(tokens.iterator().next());
+              Map<String, Integer> suggestions = result.get(spellOpts.tokens.iterator().next());
```

Comment:
All of thse `spellOpts.tokens` changes are to make sure that suggestions are checked for the right word. Previously a lot of these tests were iterating on the wrong variable so would get null suggestions. This was just a misuse in the test.